lib/plugins/plugin_api.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+# Plugin Class Extension for API Type Plugins
+class Plugin
```

Comment:
Just laziness for now, I plan on breaking out more items into their own classes. I really need to whiteboard some of the flows to find out what works best (methods, classes, etc...). About due for another refactor probably as we keep adding features.